src/peft/tuners/lora/layer.py

```diff
@@ -195,6 +197,16 @@ def update_layer(
         # Actual trainable parameters
         self.lora_A[adapter_name] = nn.Linear(self.in_features, r, bias=False)
         self.lora_B[adapter_name] = nn.Linear(r, self.out_features, bias=lora_bias)
+        if is_tied:
+            if not tied_adapters:
+                raise RuntimeError("Layer is marked as tied, but tied adapters are not provided")
+
+            lora_A_params = tied_adapters["lora_A"]
+            lora_B_params = tied_adapters["lora_B"]
+
+            self.lora_A[adapter_name].weight = torch.nn.Parameter(lora_A_params)
+            self.lora_B[adapter_name].weight = torch.nn.Parameter(lora_B_params)
```

Comment:
Let's add a comment here that weight tying is only implemented for `Embedding` => `Linear`, which is why this code is not necessary to be added to the other layer types (say, the conv layers).